### ISSUE TITLE
Misleading error message

### ISSUE THREAD

**Issue by c-git** [Canop/bacon#279]
Hi,

Thank you very much for this tool. I found a problem but not sure how feasible it is to fix it.

The problem is that if you have a TOML file where a command section is missing it gives a correct error message but points to an "incorrect" line. The line indicated actually causes the error message to overall be confusing. 

Example TOML  file missing the command

```TOML
# This is a configuration file for the bacon tool
#
# Bacon repository: https://github.com/Canop/bacon
# Complete help on configuration: https://dystroy.org/bacon/config/
# You can also check bacon's own bacon.toml file
#  as an example: https://github.com/Canop/bacon/blob/main/bacon.toml

default_job = "check"

[jobs.check2]
command = ["cargo", "check", "--color", "always"]
need_stdout = false

[jobs.check.env]
TEST_LOG = "true"
```

Error message referencing the start of the file (I acknowledge that technically there is no suitable line but maybe the line with `[jobs.check-all.env]` would make more sense if that were to be possible)

```
Error: Failed to parse configuration file at "/home/one/crate/bacon.toml"

Caused by:
    TOML parse error at line 1, column 1
      |
    1 | # This is a configuration file for the bacon tool
      | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    missing field `command`
```

**Comment by Canop** [comment]
Hum.

I pasted your example in a `bacon.toml` file, ran bacon, got this:
![image](https://github.com/user-attachments/assets/12572936-e743-4216-abf8-0dc3439f8928)

Are you sure this is the exact file you're using ? What's your version of bacon ?

**Comment by c-git** [comment]
**TLDR** Upgrade fixed it sorry for not thinking of upgrading first.

> Are you sure this is the exact file you're using ?

Yes I'm sure it's the correct file. Created a new empty file, copied it in, tried again, and got the same thing.

>  Didn't you simplify it too much for the issue ?

I'd simplified it down and kept tested to ensure I got the same error.

> What's your version of bacon ?

I had version `bacon 2.14.1` upgraded to `bacon 3.6.0` and now I get the better error message that you get. Sorry for not including the version number and for not thinking of the upgrade that it might already be solved. I'd looked for previous issues but didn't find any. Thanks for the help and sorry for the noise.
